File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        def lh(nums,target):
            low=0
            high=len(nums)-1
            while low<=high:
                mid=low+(high-low)//2
                if nums[mid]<target:
                    low=mid+1
                else:
                    high=mid-1
            return low
        def rh(nums,target):
            low=0
            high=len(nums)-1
            while low<=high:
                mid=low+(high-low)//2
                if nums[mid]<=target:
                    low=mid+1
                else:
                    high=mid-1
            return high
        lhs=lh(nums,target)
        rhs=rh(nums,target)
        if lhs<=rhs:
            return [lhs,rhs]
        return [-1,-1]



                  
        
        
        

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        int lb = lower_bound(nums.begin(), nums.end(),target) - nums.begin();
        int up = upper_bound(nums.begin(), nums.end(), target) - nums.begin();
        int n = nums.size();
        if(lb == n || nums[lb] != target){
            return {-1,-1};
        }
        else{
            return {lb, up-1};
        }
    }
};